code/modules/mob/living/carbon/xenomorph/xeno_defines.dm

```diff
@@ -8,6 +8,11 @@
 	var/caste_type_path = null
 
 	var/ancient_message = ""
+	///primordial message that is shown when a caste becomes primordial
+	var/primordial_message = ""
+
+	///name of primordial upgrade this caste looks for, keep this as define
+	var/primordial_upgrade_name = ""
```

Comment:
no need, you can check if a primordial without this

Reply:
needed for fetching the upgrade from the byname list